File: 34-find-first-and-last-position-of-element-in-sorted-array/find-first-and-last-position-of-element-in-sorted-array.java
class Solution {
    public int[] searchRange(int[] nums, int target) {
        int n = nums.length;
        int[] res = {-1,-1};
        //First position
        int beg=0,end=n-1;
        while(beg<=end){
            int mid = (beg+end)/2;
            if(nums[mid]==target){
                res[0] = mid;
                end = mid-1;
            }
            else if(target>nums[mid]){
                beg = mid+1;
            }
            else{
                end = mid-1;
            }
        }

        //Last position
        beg=0;end=n-1;
        while(beg<=end){
            int mid = (beg+end)/2;
            if(nums[mid]==target){
                res[1] = mid;
                beg = mid+1;
            }
            else if(target>nums[mid]){
                beg = mid+1;
            }
            else{
                end = mid-1;
            }
        }
        return res;
    }
}

File: 34-find-first-and-last-position-of-element-in-sorted-array/find-first-and-last-position-of-element-in-sorted-array.py
class Solution:
    def searchRange(self, nums: List[int], target: int) -> List[int]:
        res = [-1,-1]
        #1st occurrence
        low,high = 0,len(nums)-1
        while low<=high:
            mid = low + (high-low)//2
            if nums[mid]==target:
                res[0] = mid
                high = mid-1
            elif nums[mid]<target:
                low = mid+1
            else:
                high = mid-1
        #last occurrence
        low,high = 0,len(nums)-1
        while low<=high:
            mid = low + (high-low)//2
            if nums[mid]==target:
                res[1] = mid
                low = mid+1
            elif nums[mid]<target:
                low = mid+1
            else:
                high = mid-1
        return res
        